modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/milker.dm

```diff
@@ -0,0 +1,319 @@
+/// Volume 'light' mode extracts per action
+#define VOLUME_LIGHT 1
+/// Volume 'strong' mode extracts per action
+#define VOLUME_STRONG 5
+/// 'light' mode
+#define HANDLE_LIGHT 1
+/// 'strong' mode
+#define HANDLE_STRONG 2
+/// The lowest numbered setting currently available (so we dont need yet another list)
+#define HANDLE_MODE_LOWEST 1
+/// The highest numbered setting currently available
+#define HANDLE_MODE_HIGHEST 2
+/// How much pleasure is gained through 'light' mode
+#define PLEASURE_LIGHT 1
+/// How much pleasure is gained through 'strong' mode
+#define PLEASURE_STRONG 3
+/// How much arousal is gained through 'light' mode
+#define AROUSAL_LIGHT 1
+/// How much arousal is gained through 'strong' mode
+#define AROUSAL_STRONG 3
+/// How much pain is gained through 'light' mode
+#define PAIN_LIGHT 3
+
+#define PAIN_STRONG 10 // How much pain is gained through 'strong' mode// Self-(and otherwise) Expression (of mammary organs)
+
+/// The expression device
+/obj/item/milker
+	name = "milker"
+	desc = "You and me, baby, ain't nothin' but mammals."
+	icon_state = "latexballon"
+	inhand_icon_state = "nothing"
+	force = 0
+	throwforce = 0
+	item_flags = DROPDEL | ABSTRACT | HAND_ITEM
+	attack_verb_continuous = list("slaps")
+	attack_verb_simple = list("slap")
+	hitsound = 'sound/effects/snap.ogg'
+	/// Amount of pleasure inflicted from an action
+	var/pleasure_amount = PLEASURE_LIGHT
+	/// Amount of arousal inflicted from an action
+	var/arousal_amount = AROUSAL_LIGHT
+	/// Amount of pain inflicted from an action
+	var/pain_amount = PAIN_LIGHT // *chomp*
+	/// How aggressively we're going to milk someone
+	var/handle_mode = HANDLE_LIGHT
+	/// How much we're trying to express per squeeze / suck / CHOMP
+	var/squirt_volume = VOLUME_LIGHT
+
+/obj/item/milker/Initialize(mapload)
+	. = ..()
+	toggle_mode(null, TRUE)
+
+/obj/item/milker/attack(mob/living/M, mob/living/carbon/human/user)
+	if(!in_range(M, user))
+		to_chat(user, span_warning("[M] is too far away!"))
+		return FALSE
+	var/mob/living/carbon/being_milked = M
+
+	var/obj/item/organ/genital/breasts/breasts = being_milked.getorganslot(ORGAN_SLOT_BREASTS)
+	if(!breast_check(being_milked, user, breasts))
+		return FALSE
+
+	/// What's in our other hand? If it's a reagent container, we could try bottling it!
+	/// Holding a reagent container implies wanting to *use* the container, so if the container doesn't work, don't proceed!
+	var/obj/item/reagent_containers/milk_bottle = user.get_inactive_held_item()
+	if(istype(milk_bottle))
+		bottle_milk(being_milked, user, breasts, milk_bottle)
+	else // Or just drink it
+		drink_milk(being_milked, user, breasts)
+
+/// Set how vigorously you want to make the stuff come out
+/obj/item/milker/attack_self(mob/user)
+	toggle_mode(user, FALSE)
+
+/// Cycles through all the modes, looping around once it gets to the end
```

Comment:
Check the autodoc guide for autodoc'ing procs: https://github.com/Skyrat-SS13/Skyrat-tg/blob/master/.github/guides/AUTODOC.md